packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/internal/CredentialImpl.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.mongodb.internal
+
+import io.realm.internal.interop.NativePointer
+import io.realm.internal.interop.RealmInterop
+import io.realm.mongodb.Credentials
+
+internal open class CredentialImpl(
```

Comment:
We should probably provide a nice `toString()` method for debugging. In Java there is a `asJson()` method. I think that was mostly added because it was hard to keep it package protected, but it also serves as a nice debugging facility. 

With Kotlin we have `internal`, but debugging still seems a valid use case?

Reply:
In Java the `toJson` was dispatching to the C++ implementation (`credentials->serialize_as_json()`). Do we need something similar to have a standardized output. If so, then we need to expose the method in the C-API.

Reply:
Don't know if the java-identifiers for these have anything to do with the json-representation, but the representation towards the C-API has changed to use an enumeration, so don't know what to pretty-print until we have a `asJson` in the C-API. I will skip this for now, but left at TODO.

Reply:
Looking into the implementation. Prettyprinting the info does require us going to C++, so probably not worth it for now. Keeping this comment for now seems fine 👍